Packs/ctf01/Integrations/CortexXDRIRCTF/CortexXDRIRCTF.py

```diff
@@ -0,0 +1,5981 @@
+                         'or single string')
+
+    client.insert_cef_alerts(alerts)
+
+    return (
+        'Alerts inserted successfully',
+        None,
+        None
+    )
+
+
+def sort_all_list_incident_fields(incident_data):
+    """Sorting all lists fields in an incident - without this, elements may shift which results in false
+    identification of changed fields"""
+    if incident_data.get('hosts', []):
+        incident_data['hosts'] = sorted(incident_data.get('hosts', []))
+        incident_data['hosts'] = [host.upper() for host in incident_data.get('hosts', [])]
+
+    if incident_data.get('users', []):
+        incident_data['users'] = sorted(incident_data.get('users', []))
+        incident_data['users'] = [user.upper() for user in incident_data.get('users', [])]
+
+    if incident_data.get('incident_sources', []):
+        incident_data['incident_sources'] = sorted(incident_data.get('incident_sources', []))
+
+    if incident_data.get('alerts', []):
+        incident_data['alerts'] = sort_by_key(incident_data.get('alerts', []), main_key='alert_id', fallback_key='name')
+        reformat_sublist_fields(incident_data['alerts'])
+
+    if incident_data.get('file_artifacts', []):
+        incident_data['file_artifacts'] = sort_by_key(incident_data.get('file_artifacts', []), main_key='file_name',
+                                                      fallback_key='file_sha256')
+        reformat_sublist_fields(incident_data['file_artifacts'])
+
+    if incident_data.get('network_artifacts', []):
+        incident_data['network_artifacts'] = sort_by_key(incident_data.get('network_artifacts', []),
+                                                         main_key='network_domain', fallback_key='network_remote_ip')
+        reformat_sublist_fields(incident_data['network_artifacts'])
+
+
+def sync_incoming_incident_owners(incident_data):
+    if incident_data.get('assigned_user_mail') and demisto.params().get('sync_owners'):
+        user_info = demisto.findUser(email=incident_data.get('assigned_user_mail'))
+        if user_info:
+            demisto.debug(f"Syncing incident owners: XDR incident {incident_data.get('incident_id')}, "
+                          f"owner {user_info.get('username')}")
+            incident_data['owner'] = user_info.get('username')
+
+        else:
+            demisto.debug(f"The user assigned to XDR incident {incident_data.get('incident_id')} "
+                          f"is not registered on XSOAR")
+
+
+def handle_incoming_user_unassignment(incident_data):
+    incident_data['assigned_user_mail'] = ''
+    incident_data['assigned_user_pretty_name'] = ''
+    if demisto.params().get('sync_owners'):
+        demisto.debug(f'Unassigning owner from XDR incident {incident_data.get("incident_id")}')
+        incident_data['owner'] = ''
+
+
+def handle_incoming_closing_incident(incident_data):
+    closing_entry = {}  # type: Dict
+    if incident_data.get('status') in XDR_RESOLVED_STATUS_TO_XSOAR:
```

Comment:
Do we use all the XDR activity in this integration? Please remove all unused code in here.

Reply:
Not using this part so I didn't touch it.
per our discussion, I'll keep the non-used code as it is the same code duplicated from XDR, however, will add a short comment on my code to annotate its purpose (As we also removed the un-required commands from the yml as well).